.github/workflows/cimain.yaml

```diff
@@ -0,0 +1,189 @@
+name: Continuous Integration
```

Comment:
I think maintaining two almost-identical files will not very practical. Is it not possible to have a different list of commands based on the branch or some other variable?